### PR TITLE
Added Api contract test case for server server capabilities endpoint

### PR DESCRIPTION
**Pull request by Kashatlast2**
## Which Traffic Control components are affected by this PR?

- Traffic Ops Test

## What is the best way to verify this PR?
Step 1 : Follow steps from README file(traffic_ops/testing/api_contract) to run api-contract-tests.

Step 2 : To execute API Contract Testcase , navigate to traffic_ops/testing/api_contract/v4
pytest --to-user Username --to-password Password --to-url URL --config Config Path --request-template Request template --response-template Response template

Note: Config, Request template and response template paths are optional.

## PR submission checklist
- [x] This PR has tests 

- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: traffic_control/clients/python/trafficops/tosession.py [hunk 1]
```diff
@@ -1522,6 +1522,38 @@ def delete_multiple_servers_capabilities(self, data=None):
 		:raises: Union[LoginError, OperationError]
 		"""
 
+	#
+	# Server_server_capabilities
+	#
+	@api_request('get', 'server_server_capabilities', ('3.0', '4.0', '4.1', '5.0'))
+	def get_server_server_capabilities(self, query_params=none):
```
Comment (TerekhovaKate) [inline traffic_control/clients/python/trafficops/tosession.py:1529]: query_params=none: check language syntax: -> None

File: traffic_ops/testing/api_contract/v4/test_server_server_capabilities.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	logger.info(
+		"Accessing /server server capabilities endpoint through Traffic ops session.")
+
+	server_id = server_server_capabilities_post_data.get("serverId")
+	if not isinstance(server_id, int):
+		raise TypeError(
+			"malformed server server capabilities in prerequisite data; 'serverId' not an integer")
+
+	server_server_capabilities_get_response: tuple[
+		Union[dict[str, object],
```
Comment (TerekhovaKate) [inline traffic_ops/testing/api_contract/v4/test_server_server_capabilities.py:53]: please check spaces.

File: traffic_control/clients/python/trafficops/tosession.py [hunk 3]
```diff
@@ -1522,6 +1522,38 @@ def delete_multiple_servers_capabilities(self, data=None):
...
+		:ref:`to-api-server_server_capabilities`
+		:rtype: Tuple[Dict[str, Any], requests.Response]
+		:raises: Union[LoginError, OperationError]
+		"""
+
+
+	@api_request('post', 'server_server_capabilities', ('3.0', '4.0', '4.1', '5.0'))
+	def associate_server_capability_to_server(self, data=None):
+		"""
+		Associate server capability to server.
```
Comment (TerekhovaKate) [inline traffic_control/clients/python/trafficops/tosession.py:1541]: missing annotation for data param 
`:param data: The parameter data to .......... `
:type data: Dict[str, Any]

**Review by TerekhovaKate — CHANGES_REQUESTED** [review]

File: traffic_control/clients/python/trafficops/tosession.py (hunk 1 above)
Comment (Kashatlast2) [reply traffic_control/clients/python/trafficops/tosession.py]: Updated to None.

File: traffic_control/clients/python/trafficops/tosession.py (hunk 3 above)
Comment (Kashatlast2) [reply traffic_control/clients/python/trafficops/tosession.py:1541]: added annotation for data param

File: traffic_ops/testing/api_contract/v4/test_server_server_capabilities.py (hunk 2 above)
Comment (Kashatlast2) [reply traffic_ops/testing/api_contract/v4/test_server_server_capabilities.py]: Fixed spacing.

File: traffic_ops/testing/api_contract/v4/conftest.py [hunk 4]
```diff
@@ -1859,3 +1859,36 @@ def static_dns_entries_data_post(to_session: TOSession, request_template_data: l
...
+	server_capability = server_server_capabilities["serverCapability"]
+
+	logger.info("New server_server_capabilities data to hit POST method %s", server_server_capabilities)
+
+	# Hitting server server capabilities POST method
+	response: tuple[JSONData, requests.Response] = to_session.associate_server_capability_to_server(server_id=server_id, data=server_server_capabilities)
+	resp_obj = check_template_data(response, "server_server_capabilities")
+	yield resp_obj
+	server_id = resp_obj.get("serverId")
+	msg = to_session.delete_server_capability_association_to_server(query_params=server_id, server_capability=server_capability)
```
Comment (gokulakrishnansvm) [inline traffic_ops/testing/api_contract/v4/conftest.py:1889]: This won't delete the data, You are passing two arguments query_params and server_capability,
But the delete method in tosession accepts only query_params,
`delete_server_capability_association_to_server(self, query_params=None)`
Just pass the query_params data itself as,
`msg = to_session.delete_server_capability_association_to_server(query_params = {"serverId":server_id, "serverCapability":server_capability})`

Comment (Kashatlast2) [reply traffic_ops/testing/api_contract/v4/conftest.py:1889]: Yes this is true. I made this change. I'm not sure why it didn't push with the updated code.

File: traffic_ops/testing/api_contract/v4/conftest.py [hunk 5]
```diff
@@ -1093,10 +1093,15 @@ def server_data_post(to_session: TOSession, request_template_data: list[JSONData
 	type_object = create_or_get_existing(to_session, "types", "type", type_data,
 				      {"useInTable": "server"})
 	type_id = type_object["id"]
-	server["typeId"] = type_id
+	edge_type_id = pytestconfig.cache.get("edgeTypeId", default=None)
+	logger.info("Here %s", edge_type_id)
```
Comment (ericholguin) [inline traffic_ops/testing/api_contract/v4/conftest.py:1097]: you can remove debug logging

Comment (Kashatlast2) [reply traffic_ops/testing/api_contract/v4/conftest.py]: Removed

**Pull request merged**